lib/fluent/diagtool/collectutils.rb

```diff
@@ -422,9 +422,17 @@ def match_platform?(platforms_option)
 
     def collect_bundled_plugins
       gemfile_path = if fluent_package?
-                       "/opt/fluent/share/Gemfile"
+                       if Diagtool.windows?
+                         File.join(ENV["FLUENT_PACKAGE_TOPDIR"], "share/Gemfile")
+                       else
+                         "/opt/fluent/share/Gemfile"
+                       end
                      else
-                       "/opt/td-agent/share/Gemfile"
+                       if Diagtool.windows?
+                         File.join(ENV["TD_AGENT_TOPDIR"], "share/Gemfile")
+                       else
+                         "/opt/td-agent/share/Gemfile"
+                       end
                      end
```

Comment:
This class should not have platform-specific logic.
How about fixing `lib/fluent/diagtool/windows/collectutils.rb`.

For example, we can separate the `gemfile_path` logic to a function:

`lib/fluent/diagtool/collectutils.rb`

```rb
    def collect_bundled_plugins
      File.read(gemfile_path).lines(chomp: true).grep(/\Agem "fluent-plugin-/).collect do |line|
        gem_name = line[/\Agem "(fluent-plugin-.+?)"/, 1]
        platforms_option = line[/platforms: (\w+)/, 1]
        next gem_name, platforms_option
      end.select do |gem_name, platforms_option|
        platforms_option.nil? || match_platform?(platforms_option)
      end.collect do |gem_name, platforms_option|
        gem_name
      end
    end

    (...)

    def gemfile_path
      if fluent_package?
        "/opt/fluent/share/Gemfile"
      else
        "/opt/td-agent/share/Gemfile"
      end
    end
```

And, we can add Windows-specific logic to `lib/fluent/diagtool/windows/collectutils.rb`

```rb
      def gemfile_path
        if fluent_package?
          File.join(ENV["FLUENT_PACKAGE_TOPDIR"], "share/Gemfile")
        else
          File.join(ENV["TD_AGENT_TOPDIR"], "share/Gemfile")
        end
      end
```

Reply:
Thanks!